src/main.ts

```diff
@@ -62,11 +84,36 @@ export async function runSorald(
   return allRepairs;
 }
 
+function filterViolationSpecsByRatchet(
```

Comment:
You forgot to add docstrings here. :)

Reply:
As this function isn't exported, I figured it's like "private" in TypeScript, and therefore omitted the docstring.

Reply:
According to meeting: add explanatory docstring.